src/server/mod.ts

```diff
@@ -86,10 +85,51 @@ export async function start(routes: Manifest, opts: StartOptions = {}) {
     };
   }
 
+  const portEnv = Deno.env.get("PORT");
+  if (portEnv !== undefined) {
+    opts.port ??= parseInt(portEnv, 10);
+  }
+
+  const handler = ctx.handler();
+
+  if (opts.port) {
+    await bootServer(handler, opts);
+  } else {
+    // No port specified, check for a free port. Instead of picking just
+    // any port we'll check if the next one is free for UX reasons.
+    // That way the user only needs to increment a number when running
+    // multiple apps vs having to remember completely different ports.
+    let firstError;
+    for (let port = 8000; port < 8020; port++) {
+      try {
+        await bootServer(handler, { ...opts, port });
+        firstError = undefined;
+        break;
+      } catch (err) {
+        if (err.code === "EADDRINUSE") {
```

Comment:
This is correct - but `err instanceof Deno.errors.AddrInUse` is more Deno idiomatic.